### DISCUSSION TITLE
Most SO services won't start

### DISCUSSION THREAD

**Discussion by netsecninja** [Security-Onion-Solutions/securityonion#1689] [Unsupported Versions] [answered]
Greetings all! I'm looking for some troubleshooting assistance. I upgraded to 2.3.2, and after a reboot, I get this:

![image](https://user-images.githubusercontent.com/10831043/97370611-ff187d00-1874-11eb-9046-6f2c8174f66a.png)

I left it overnight to see if time would help, nope. I have rebooted a couple times, no change. When I try to start some of the errored services with so-SERVICENAME-start, I get this in the console:

`salt.exceptions.CommandExecutionError: Error 404: no such image: securityonion:5000/securityonion/so-thehive-cortex:2.3.1: No such image: securityonion:5000/securityonion/so-thehive-cortex:2.3.1`

This appears to indicate the module wasn't updated correctly, since it's not 2.3.2. I even tried running soup again to see if it would catch the older modules, but it said I was already up to date. Some modules would manually start, others would not. Here's the list that would not:

- cortex
- strelka
- thehive-es

After starting all the other services by hand, I rebooted to see if that made a difference. The services that had to be started manually did start on their own, but the ones listed above were still erroring. I'm not sure what else to do to troubleshoot this. Any ideas on something more I can check?

**Comment by greatapoc** [comment]
I'm not a dev but try so-docker-refresh.

**Reply by netsecninja** [reply]
Good suggestion! However, I just tried it and those services all listed "Layer already exists". When it completed I tried to start them manually, and got the same error showing 2.3.1.

**Reply by stedaniels** [reply]
Is this repeatable on a reinstall?

**Reply by netsecninja** [reply]
I don't think we are at the nuke from orbit step just yet. But hopefully I'm the only one with this issue, as I assume many others have upgraded to 2.3.2 without issues.

**Comment by shapesh** [comment]
Try restarting all the containers
docker restart $(docker ps -a -q)

**Comment by Masaya-A** [comment, marked as answer]
All of my rigs which were updated from 2.3.1 or RC3 became similar situation.

Following way to solve for all of my rigs:
1. Delete untagged docker containers. (ie. Old Version of images of the containers were deleted during update, but containers weren't deleted)
$ sudo docker ps -a
$ sudo docker rm -f <Container ID 1> <Container ID 2>....
2. Run so-docker-refresh.
$ sudo so-docker-refresh
3. Reboot your rig.
$ sudo reboot

**Reply by netsecninja** [reply]
Your suggestion worked, thanks! For reference, here's the container IDs I deleted, they were the only ones without an image name and were in an exited state:
495ff0ff1f46
672dca479174
51f4f2809f5e
79ac6beeb690
f6ce1e61ba49

**Comment by jertel** [comment]
If you're still in this bad state please share your `docker images --digests` output.

**Comment by schmug** [comment]
edit: I'm going to open a new discussion because I noticed it was answered.

I thought I would jump in, I can open a new discussion if you'd like though.

On a new ISO install, three times, I am getting ERRORs on various services.

![image](https://user-images.githubusercontent.com/38227427/97446049-53b70900-1904-11eb-9c4b-273e077ccbdc.png)

`securityonion/so-strelka-manager                           2.3.2               sha256:ea94ea86dd3d3197c316b24fcd65f85077cdd2fa4e654e260bcbf0fb1a00876f   6e4bb2ca8a38        7 days ago          20.8MB
securityonion-1:5000/securityonion/so-strelka-manager      2.3.2               sha256:ea94ea86dd3d3197c316b24fcd65f85077cdd2fa4e654e260bcbf0fb1a00876f   6e4bb2ca8a38        7 days ago          20.8MB
securityonion-1:5000/securityonion/so-strelka-frontend     2.3.2               sha256:c7a9adeb84f6675d43f13a13c667fa303e9b23ea70bfc178d728660b0f334de5   7e27eee1f26d        7 days ago          21.3MB
securityonion/so-strelka-frontend                          2.3.2               sha256:c7a9adeb84f6675d43f13a13c667fa303e9b23ea70bfc178d728660b0f334de5   7e27eee1f26d        7 days ago          21.3MB
securityonion/so-strelka-filestream                        2.3.2               sha256:3121cee3101343629b2918481d97193b8f9975923607e7a1d3a8118b7186391a   fdc5cfd98a58        8 days ago          20MB
securityonion-1:5000/securityonion/so-strelka-filestream   2.3.2               sha256:3121cee3101343629b2918481d97193b8f9975923607e7a1d3a8118b7186391a   fdc5cfd98a58        8 days ago          20MB
securityonion/so-elastalert                                2.3.2               sha256:37225d521e795e4817b260cd8749d33c1de1955c556acab0192dd5553644bec3   1a5b57a35105        8 days ago          145MB
securityonion-1:5000/securityonion/so-elastalert           2.3.2               sha256:37225d521e795e4817b260cd8749d33c1de1955c556acab0192dd5553644bec3   1a5b57a35105        8 days ago          145MB
securityonion/so-playbook                                  2.3.2               sha256:054049476826732317fe2edce9dd2992ec83cf949f6cff7620cfadc64c8a61b4   c0ad97006462        8 days ago          629MB
securityonion-1:5000/securityonion/so-playbook             2.3.2               sha256:054049476826732317fe2edce9dd2992ec83cf949f6cff7620cfadc64c8a61b4   c0ad97006462        8 days ago          629MB`

I have `docker remove` on each of the ERROR status containers and `so-docker-refresh` and `reboot` to no avail. 

Also of note is mysql shows "unhealthy"

I chose defaults for all the GUI steps, but setup ended with this: `Errors detected during setup; skipping post-setup steps to allow for analysis of failures.`

```
Error response from daemon: Container daee7a167469a52286a7372319ab065948f7cf6df510b3e4de687531b52941ff is not running
[ERROR   ] Command '['docker', 'exec', 'so-fleet', 'fleetctl', 'get', 'enroll-secret', 'default']' failed with return code: 1
[ERROR   ] stdout: Error response from daemon: Container daee7a167469a52286a7372319ab065948f7cf6df510b3e4de687531b52941ff is not running
[ERROR   ] retcode: 1
[ERROR   ] Command 'docker exec so-fleet fleetctl get enroll-secret default' failed with return code: 1
[ERROR   ] output: Error response from daemon: Container daee7a167469a52286a7372319ab065948f7cf6df510b3e4de687531b52941ff is not running
 </html>nter>nginx/1.17.10</center>ror</h1></center>
[INFO    ] Executing command 'yum check-update 2>&1 | grep "Error: rpmdb open failed"' in directory '/root'
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.
```

**Reply by shapesh** [reply]
I also had the same issue 3 times with grafana, kibana and suricata.

sudo so-docker-refresh
sudo docker restart $(docker ps -a -q)   //to restart all the containers.

alternatively you can do
docker ps -a 
to check the containers that are down and restart each container by
docker restart (container_id)

**Reply by schmug** [reply]
Thanks for the reply, I have been able to get some of the containers up like this but haven't been able to get elastalert or mysql working.

**Reply by schmug** [reply]
With 2.3.10, I've had a distributed install succeed.
